pkg/client/query/supplierquerier.go

```diff
@@ -44,6 +51,16 @@ func (supq *supplierQuerier) GetSupplier(
 	ctx context.Context,
 	operatorAddress string,
 ) (sharedtypes.Supplier, error) {
+	logger := supq.logger.With("query_client", "supplier", "method", "GetSupplier")
+
+	// Check if the supplier is present in the cache.
+	if supplier, found := supq.suppliersCache.Get(operatorAddress); found {
+		logger.Debug().Msgf("cache hit for key: %s", operatorAddress)
+		return supplier, nil
+	}
+
+	logger.Debug().Msgf("cache miss for key: %s", operatorAddress)
```

Comment:
Leaving one comment but please update everywhere. If these logs ever becomes the source for debugging, you want it to be ULTRA obvious.

```suggestion
	logger.Debug().Msgf("cache miss for operator address key: %s", operatorAddress)
```